src/app/content/components/Page/contentDOMTransformations.ts

```diff
@@ -130,26 +86,25 @@ function fixLists(rootEl: HTMLElement) {
   });
 }
 
-// hide text visually, but make rangy still search through it
-// when highlighting a search result
-// https://github.com/timdown/rangy/wiki/Text-Range-Module#visible-text
-const initialSolutionStyles = 'display: block; overflow: hidden; height: 0px';
-
 function wrapSolutions(rootEl: HTMLElement, intl: IntlShape) {
   const title = intl.formatMessage({id: 'i18n:content:solution:toggle-title'});
-  const showSolutionText = intl.formatMessage({id: 'i18n:content:solution:show'});
 
-  // Wrap solutions in a div so "Show/Hide Solutions" work
+  // Wrap solutions in a details element so "Show/Hide Solutions" work
   rootEl.querySelectorAll('.exercise .solution, [data-type="exercise"] [data-type="solution"]').forEach((el) => {
-    el.setAttribute('aria-label', showSolutionText);
-    el.setAttribute('aria-expanded', 'false');
+    el.setAttribute('aria-label', intl.formatMessage({id: 'i18n:content:solution:toggle-title'}));
     const contents = el.innerHTML;
-    el.innerHTML = `
-      <div class="ui-toggle-wrapper">
-        <button class="btn-link ui-toggle" title="${title}" data-content="${showSolutionText}"></button>
-      </div>
-      <section class="ui-body" role="alert" style="${initialSolutionStyles}">${contents}</section>
+    const detailsEl = document!.createElement('details');
+
+    Array.from(el.attributes).forEach((attr) => {
+      detailsEl.setAttribute(attr.name, attr.value);
+    });
+
+    detailsEl.classList.add('ui-toggle-wrapper');
```

Comment:
Is this still necessary?

Reply:
I think you're right, after all the changes, I thought some CSS remained that targeted that class, but I just searched and can't find anything. I'll remove it and re-test to make sure.